src/realm/sync/instruction_applier.cpp

```diff
@@ -226,11 +226,8 @@ void InstructionApplier::operator()(const Instruction::CreateObject& instr)
                 }
                 m_last_object = table->create_object_with_primary_key(id);
             },
-            [&](GlobalKey key) {
-                if (pk_col) {
-                    bad_transaction_log("CreateObject(GlobalKey) on table with a primary key");
-                }
-                m_last_object = table->create_object(key);
+            [&](GlobalKey) {
```

Comment:
I guess we don't remove this type all together so there is no breaking change?

Reply:
Well, this is kind of breaking in that we don't support GlobalKey anymore. But is was never used in context of MongoDB cloud. I think next step would be to remove the type altogether.

Reply:
what's stopping us to do it now?